src/main/kotlin/uk/gov/justice/digital/hmpps/subjectaccessrequestworker/client/PrisonApiClient.kt

```diff
@@ -1,37 +1,61 @@
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
-import org.springframework.web.reactive.function.client.WebClientRequestException
-import org.springframework.web.reactive.function.client.WebClientResponseException
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.events.ProcessingEvent.ACQUIRE_AUTH_TOKEN
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.events.ProcessingEvent.GET_OFFENDER_NAME
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.exception.FatalSubjectAccessRequestException
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.models.SubjectAccessRequest
+import uk.gov.justice.digital.hmpps.subjectaccessrequestworker.utils.WebClientRetriesSpec
 import java.util.Locale
 
 @Service
 class PrisonApiClient(
   private val prisonApiWebClient: WebClient,
+  private val webClientRetriesSpec: WebClientRetriesSpec,
 ) {
 
-  fun getOffenderName(subjectId: String): String {
+  fun getOffenderName(subjectAccessRequest: SubjectAccessRequest, subjectId: String): String {
     return try {
       val response = prisonApiWebClient
         .get()
         .uri("/api/offenders/$subjectId")
         .retrieve()
+        .onStatus(
+          webClientRetriesSpec.is4xxStatus(),
+          webClientRetriesSpec.throw4xxStatusFatalError(
+            GET_OFFENDER_NAME,
+            subjectAccessRequest,
+            mapOf("subjectId" to subjectId),
+          ),
+        )
         .bodyToMono(String::class.java)
+        .retryWhen(
+          webClientRetriesSpec.retry5xxAndClientRequestErrors(
+            GET_OFFENDER_NAME,
+            subjectAccessRequest,
+            mapOf("subjectId" to subjectId),
+          ),
+        )
         .block()
 
       val details = JSONParser(response).parseObject()
-      "${details.get("lastName").toString().uppercase()}, ${
-        details.get("firstName").toString().lowercase()
+      "${details["lastName"].toString().uppercase()}, ${
+        details["firstName"].toString().lowercase()
           .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString() }
       }"
-    } catch (exception: WebClientRequestException) {
-      throw RuntimeException("Connection to ${exception.uri.authority} failed.")
-    } catch (exception: WebClientResponseException.ServiceUnavailable) {
-      throw RuntimeException("${exception.request?.uri?.authority} is unavailable.")
-    } catch (exception: WebClientResponseException.Unauthorized) {
-      throw RuntimeException("Invalid credentials used.")
+    } catch (ex: ClientAuthorizationException) {
+      throw FatalSubjectAccessRequestException(
+        message = "prisonApiClient error authorization exception",
+        cause = ex,
+        event = ACQUIRE_AUTH_TOKEN,
+        subjectAccessRequestId = subjectAccessRequest.id,
+        params = mapOf(
+          "cause" to ex.cause?.message,
+        ),
+      )
```

Comment:
Removed other catch blocks as any exception will be caught and logged in the top level handler.